onnxscript/type_annotation.py

```diff
@@ -146,3 +164,102 @@ def get_return_types(typeinfo: type | typing.Sequence[type]) -> typing.Sequence[
     if get_origin(typeinfo) is tuple:
         return get_args(typeinfo)
     return (typeinfo,)
+
+
+# A sorted list of all type strings used in an OpSchema
+ALL_TYPE_STRINGS = (
+    "tensor(bfloat16)",
+    "tensor(bool)",
+    "tensor(double)",
+    "tensor(float)",
+    "tensor(float16)",
+    "tensor(int16)",
+    "tensor(int32)",
+    "tensor(int64)",
+    "tensor(int8)",
+    "tensor(string)",
+    "tensor(uint16)",
+    "tensor(uint32)",
+    "tensor(uint64)",
+    "tensor(uint8)",
+)
+
+
+def pytype_to_input_strings(pytype: TypeAnnotationValue) -> list[str]:
+    """Returns a list of all supported input types in string representation for a given type annotation.
+
+    Args:
+        pytype: A type annotation.
+
+    Returns:
+        A list of all supported input types for the given type annotation.
+        Ensures that the list is sorted in the same order as ALL_TYPE_STRINGS.
+    """
+    if pytype is None:
+        return list(ALL_TYPE_STRINGS)
+    if pytype is type(None):
+        return list(ALL_TYPE_STRINGS)
```

Comment:
Why would we see NoneType as all type? Looks like we prefer NoneType to wrong type.

Reply:
Is it for custom functions?

Reply:
Does `ALL_TENSOR_TYPE_STRINGS` sound good to you? That's more what's intended

Reply:
> Why would we see NoneType as all type? Looks like we prefer NoneType to wrong type.

When a parameter is not annotated at all, we get NoneType but it will be treated as a tensor (actually, do we just get None? I will remove NoneType to see what happens)

Reply:
Removed.